src/main/java/io/codelex/flightplanner/flights/FlightsRepository.java

```diff
@@ -0,0 +1,96 @@
+package io.codelex.flightplanner.flights;
+
+import io.codelex.flightplanner.search.PageResult;
+import io.codelex.flightplanner.search.SearchFlightsRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class FlightsRepository {
+    private final List<Flight> listOfFlights = new ArrayList<>();
+    private final List<Airport> airportList = new ArrayList<>();
+    private Long id = 0L;
+
+    public synchronized Flight addFlight(AddFlightRequest addFlightRequest) {
+        for (Flight flight : listOfFlights) {
+            if (flight.matches(addFlightRequest)) {
+                throw new ResponseStatusException(HttpStatus.CONFLICT);
+            }
+        }
+
+        if (addFlightRequest.airportsAreEqual()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        if (!addFlightRequest.datesAreCorrect()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        id++;
+        airportList.add(addFlightRequest.getFrom());
+        airportList.add(addFlightRequest.getTo());
+
+        Flight newFlight = new Flight(id, addFlightRequest);
+        listOfFlights.add(newFlight);
+        return newFlight;
+    }
+
+    public synchronized Flight getFlightById(Long id) {
+        Flight flightFound = null;
+
+        for (Flight flight : listOfFlights) {
+            if (flight.getId() == id) {
+                flightFound = flight;
+            }
+        }
+
+        if (flightFound == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+        return flightFound;
+    }
+
+    public PageResult searchFlights(SearchFlightsRequest searchFlightsRequest) {
+        List<Flight> foundedFlights = new ArrayList<>();
+
+        for (Flight flight : listOfFlights) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            String flightDepartureDate = flight.getDepartureTime().format(formatter);
```

Comment:
`flight.getDepartureTime().toLocalDate()` un labāk request pārveido uz LocalDate, būs daudz vieglāk manipulēt ar to nekā String ja vajadzēs kādas papilldus izmaiņas

Reply:
Paldies, to pārveidoju kopā ar stream ieviešanu.

Par LocalDate un LocalDateTime 🤔? Jānis minēja, lai izmantojam LocalDateTime. Bet arī LocalDate var izmantot šajā gadījumā, jo LocalDateTime, atšķirība ir nanosekundes, kuras tāpat neizmantojam.  Ko tad labāk izmantot?

Reply:
tas pilnība atkarīgs no uzdevuma nosacījumiem, ko izmantot
main point was that LocalDate and LocalDateTime būs labak par String 😄